aten/src/ATen/Storage.cpp

```diff
@@ -1,12 +1,17 @@
-#include "THStorageClass.hpp"
+#include <ATen/Storage.h>
+#include <ATen/Context.h>
 
-THStorage::THStorage(
+namespace at {
+
+Storage::Storage(
+    at::Backend backend,
     at::ScalarType scalar_type,
     ptrdiff_t size,
     at::DataPtr data_ptr,
     at::Allocator* allocator,
     char flag)
     : scalar_type(scalar_type),
+      backend_(backend),
```

Comment:
Technically, you already know the backend from the contents of `data_ptr`, which has a `Device`.